docshare.go

```diff
@@ -89,59 +96,115 @@ func main() {
 			ch <- struct{}{}
 			wg.Add(1)
 			go func(docID string) {
+				// TODO: figure out a better way to create context for each goroutine.
+				ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
```

Comment:
Why worry about timeouts only for remove but not add?

Reply:
It's just I always try to make minimal changes. In this case I only needed the new function for removal, which I thought would be better with context. The "add" part only needed some URL parameters, so adding context would've been out of scope for this change 

In fact, the PR #3 does add context to the other part.

Reply:
I have to admin though, changing log message is also out of scope for this change. I guess I should've added it separately but here we are.